pkg/artifactory/resource_artifactory_remote_repository.go

```diff
@@ -484,6 +484,11 @@ func resourceRemoteRepositoryRead(d *schema.ResourceData, m interface{}) error {
 	c := m.(*ArtClient).ArtOld
 
 	repo, resp, err := c.V1.Repositories.GetRemote(context.Background(), d.Id())
+
+	if resp != nil {
```

Comment:
The equality check should be `resp == nil` but check `err` first. If you check error first you probably never need to check if `resp` is `nil`.